nixos/modules/services/cluster/k3s/default.nix

```diff
@@ -78,7 +78,8 @@ in
 
     systemd.services.k3s = {
       description = "k3s service";
-      after = mkIf cfg.docker [ "docker.service" ];
+      after = [ "network-online.service" "firewall.service" ] ++ (if cfg.docker then [ "docker.service" ] else []);
```

Comment:
This can be ` ++ (optional cfg.docker [ "docker.service" ])`, which I think is slightly more idiomatic.

Reply:
Right I had seen that before but forgot, thanks!